monsterCatalog.py

```diff
@@ -0,0 +1,280 @@
+import flask
+import httplib2
+import json
+import requests
+import random
+import string
+import datetime
+import boto3
+import base64
+import uuid
+import werkzeug
+
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.client import FlowExchangeError
+
+import sqlalchemy
+from database import Base, User, Monster
+
+
+CLIENT_ID = json.loads(
+    open('client_secret_478230017741-rd1hon3dcgt9eoutdkjd5lsiidat2gp5.apps.googleusercontent.com.json', 'r').read())['web']['client_id']
+APPLICATION_NAME = "Monster Catalog"
+
+
+app = flask.Flask(__name__)
+
+engine = sqlalchemy.create_engine('postgres:///monstercatalog')
+Base.metadata.bind = engine
+
+DBSession = sqlalchemy.orm.sessionmaker(bind=engine)
+db_session = DBSession()
+
+
+@app.route('/')
+def main_page():
+    monsters = db_session.query(Monster).limit(20).all()
+    return flask.render_template('main.html', monsters=monsters)
+
+
+# Create anti-forgery state token
+@app.route('/signin')
+def sign_in():
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in xrange(32))
+    flask.session['state'] = state
+    return flask.render_template('signup.html', STATE=state)
+
+
+@app.route('/create', methods=['GET', 'POST'])
+def create_monster():
+    # print flask.request.form
+    if 'username' not in flask.session:
+        return flask.redirect('/signin')
+    if flask.request.method == 'POST':
+        if 'encoded_picture' in flask.request.form:
+            encoded_picture = flask.request.form.get('encoded_picture')
+            picture = base64.b64decode(str(encoded_picture))
+            picture_id = str(uuid.uuid4())
+            picture_url = 'https://s3.amazonaws.com/monster-catalog/' + picture_id
+
+            session = boto3.session.Session(
+                aws_access_key_id=app.config['AWS_ACCESS_KEY_ID'],
+                aws_secret_access_key=app.config['AWS_SECRET_ACCESS_KEY']
+            )
+            s3 = session.resource(service_name='s3')
+
+            s3.Bucket(app.config['AWS_S3_BUCKET']).put_object(
+                Key=picture_id,
+                Body=picture,
+                ContentType='image/png')
+
+            monster = Monster(
+                name=flask.request.form['name'],
+                diet=flask.request.form['diet'],
+                enjoys=flask.request.form['enjoys'],
+                creator=flask.session['user_id'],
+                picture=picture_url,
+                intentions=flask.request.form['intentions'],
+                created_date=datetime.datetime.now()
+                )
+            db_session.add(monster)
+            try:
+                db_session.commit()
+                print "committed"
```

Comment:
Generally speaking avoid "print" statements.  In this case you might well want to log something, but make sure you log it in a way that it will show up in the logs wherever it's running on a live web server.